### PR TITLE
infra: Add custom rpmlint config

### PR DESCRIPTION
**Pull request by elkoniu**
Custom rpmlint config file is merged with default config. This way we can supress some errors which are inrelevant to our project.

### CODE REVIEW

**Comment by elkoniu** [comment]
This is very `basic` but also `safe` change to default rpmlint config. It wont fix all the issues but will get rid of identified false positives (for now).

**Comment by elkoniu** [comment]
Just for the record, here is sample of current F41 dist git rpmlint run:
```
===================================================== rpmlint session starts ====================================================
rpmlint: 2.5.0
configuration:
    /usr/lib/python3.12/site-packages/rpmlint/configdefaults.toml
    /etc/xdg/rpmlint/fedora-legacy-licenses.toml
    /etc/xdg/rpmlint/fedora-spdx-licenses.toml
    /etc/xdg/rpmlint/fedora.toml
    /etc/xdg/rpmlint/scoring.toml
    /etc/xdg/rpmlint/users-groups.toml
    /etc/xdg/rpmlint/warn-on-functions.toml
checks: 32, packages: 17

anaconda.src: E: spelling-error ('metapackage', '%description -l en_US metapackage -> meta package, meta-package, prepackage')
anaconda.x86_64: E: spelling-error ('metapackage', '%description -l en_US metapackage -> meta package, meta-package, prepackage')
anaconda-dracut.x86_64: E: spelling-error ('kickstarts', '%description -l en_US kickstarts -> kick starts, kick-starts, kicks tarts')
anaconda-install-env-deps.x86_64: E: spelling-error ('metapackage', '%description -l en_US metapackage -> meta package, meta-package, prepackage')
anaconda-install-img-deps.x86_64: E: spelling-error ('metapackage', '%description -l en_US metapackage -> meta package, meta-package, prepackage')
anaconda-install-img-deps.x86_64: E: spelling-error ('iso', '%description -l en_US iso -> ISO, is, so')
anaconda-install-img-deps.x86_64: E: spelling-error ('eg', '%description -l en_US eg -> eh, e, g')
anaconda-install-img-deps.x86_64: E: spelling-error ('lorax', '%description -l en_US lorax -> borax')
anaconda-core-debuginfo.x86_64: W: only-non-binary-in-usr-lib
anaconda-core.x86_64: W: obsolete-not-provided booty
anaconda-core.x86_64: W: no-manual-page-for-binary anaconda-cleanup
anaconda-core.x86_64: W: no-manual-page-for-binary anaconda-disable-nm-ibft-plugin
anaconda-core.x86_64: W: no-manual-page-for-binary anaconda-nm-disable-autocons
anaconda-core.x86_64: W: no-manual-page-for-binary instperf
anaconda-core.x86_64: W: no-manual-page-for-binary anaconda
anaconda-core.x86_64: W: no-manual-page-for-binary handle-sshpw
anaconda.x86_64: W: no-documentation
anaconda-core.x86_64: W: no-documentation
anaconda-dracut.x86_64: W: no-documentation
anaconda-gui.x86_64: W: no-documentation
anaconda-install-env-deps.x86_64: W: no-documentation
anaconda-install-img-deps.x86_64: W: no-documentation
anaconda-tui.x86_64: W: no-documentation
anaconda-widgets.x86_64: W: no-documentation
anaconda.x86_64: E: no-binary
anaconda-core-debuginfo.x86_64: E: no-binary
anaconda-install-env-deps.x86_64: E: no-binary
anaconda-install-img-deps.x86_64: E: no-binary
anaconda.spec:1954: W: macro-in-%changelog %post
anaconda.spec:2712: W: macro-in-%changelog %anaconda
anaconda.spec:1954: W: macro-in-%changelog %post
anaconda.spec:2712: W: macro-in-%changelog %anaconda
anaconda-core.x86_64: E: explicit-lib-dependency flatpak-libs
anaconda-core.x86_64: E: explicit-lib-dependency libselinux-python3
anaconda-install-env-deps.x86_64: E: explicit-lib-dependency libblockdev-lvm-dbus
anaconda-install-img-deps.x86_64: E: explicit-lib-dependency librsvg2
anaconda-gui.x86_64: W: dangling-symlink /usr/share/anaconda/window-manager/glib-2.0/schemas/org.gnome.desktop.enums.xml /usr/share/glib-2.0/schemas/org.gnome.desktop.enums.xml
anaconda-gui.x86_64: W: dangling-symlink /usr/share/anaconda/window-manager/glib-2.0/schemas/org.gnome.desktop.wm.keybindings.gschema.xml /usr/share/glib-2.0/schemas/org.gnome.desktop.wm.keybindings.gschema.xml
anaconda-gui.x86_64: W: dangling-symlink /usr/share/anaconda/window-manager/glib-2.0/schemas/org.gnome.desktop.wm.preferences.gschema.xml /usr/share/glib-2.0/schemas/org.gnome.desktop.wm.preferences.gschema.xml
anaconda-core.x86_64: W: dangling-relative-symlink /usr/lib/.build-id/bb/8e8a1aec980e68bcfae6ac362ebae9d7c08e0a ../../../../usr/libexec/anaconda/dd_extract
anaconda-core.x86_64: W: dangling-relative-symlink /usr/lib/.build-id/f2/cceaadfee0078433b2d6dd2cb08ec49fd217bc ../../../../usr/libexec/anaconda/dd_list
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/anaconda.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/conf.d/00-do-nothing.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/conf.d/01-still-do-nothing.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/conf.d/README
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/almalinux.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/centos.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/circle.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-designsuite.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-eln.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-iot.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-kde.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-kinoite.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-sericea.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-server.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-silverblue.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora-workstation.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/fedora.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/ovirt.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/rhel.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/rhvh.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/rocky.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/scientific-linux.conf
anaconda-core.x86_64: W: conffile-without-noreplace-flag /etc/anaconda/profile.d/virtuozzo-linux.conf
============= 16 packages and 1 specfiles checked; 16 errors, 48 warnings, 96 filtered, 16 badness; has taken 2.4 s =============

```

**Review by KKoukiou — COMMENTED** [review]
How did you test it? I think you can create a dist-git PR , (which you will not merge) and let the tests run to see if the errors get whitelisted with this.

**Comment by M4rtinK** [comment]
> How did you test it? I think you can create a dist-git PR , (which you will not merge) and let the tests run to see if the errors get whitelisted with this.

Yep, this should work and we definitely used it in the past to test similar changes. :)

File: anaconda.spec.in [hunk 1]
```diff
@@ -189,6 +189,7 @@ for live installations.
 
 %package install-env-deps
 Summary: Installation environment specific dependencies
+BuildArchitectures: noarch
```
Comment (jkonecny12) [inline anaconda.spec.in:192]: I wonder, could this be a noarch package when it has arch specific conditions on requires list? Is it fine to do?

Comment (KKoukiou) [reply anaconda.spec.in:192]: Good point.
If the subpackage has architecture-specific conditions (like Requires for certain architectures), it should not be marked as noarch.

Comment (elkoniu) [reply anaconda.spec.in]: I have removed `noarch` changes and added those packages to rpmlint whitelist.

**Review by KKoukiou — APPROVED** [review]

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

**Pull request merged**